### ISSUE TITLE
refused to connect.

### ISSUE THREAD

**Issue by GitIgnoreMaybe** [paxswill/iocage-zigbee2mqtt#2]
Hey Will,

first of all big thanks for the great plugin. This saved me a lot of time.

The plugin worked perfectly up till now. My Zigbee devices (lights) just didn't react anymore and after checking HA I've seen that the zigbee2mqtt service refuses the connection suddenly. Any clue what caused that issue? After several hours I just run out of ideas.

**Comment by paxswill** [comment]
What is having trouble connecting to what? Is HA able to connect to the MQTT broker (that'll probably be mentioned in the HA logs)? If not, try checking the MQTT broker settings (and checking the logs there).

If HA can connect to the MQTT broker, is zigbee2mqtt publishing to MQTT? One way to check is to go to the "Configure" area for the MQTT integration in HA, and then listen to the `zigbee2mqtt/bridge/info` topic. If zigbee2mqtt is connected to the MQTT broker, a status message will show up pretty quickly.

Is zigbee2mqtt not able to connect to the zigbee dongle? For that I'd start by checking to see which device it's showing up as (try unplugging, then plugging it back in and seeing what new messages show up in `dmesg`). If it's showing up as something different for some reason, creating a udev rule to rename the devices to something might fix that permanently (once upon a time I knew how to do this, but not anymore. If you do need to do this, have the new names start with `cuaU`/`ttyU`, as only devices with those prefixes are passed through to the plugin jail).

**Comment by GitIgnoreMaybe** [comment]
Thanks for the quick reply. 

I checked the HA logs:
- MQTT broker is connected

I checked the MQTT broker topic ` zigbee2mqtt/bridge/info`
- I received a pingresp

I checked the dongle inside of the HA ioncage `ls -l /dev | grep cua ` by removing and plugging in the dongle.

Plugged in:
```
root@HomeAssistant:~ # ls -l /dev | grep cua
crw-rw----  1 uucp  dialer    0x3d Feb 22 02:46 cuau0
crw-rw----  1 uucp  dialer    0xca Feb 22 03:45 cuaU0
crw-rw----  1 uucp  dialer    0x3e Feb 22 02:46 cuau0.init
crw-rw----  1 uucp  dialer    0xcb Feb 22 03:45 cuaU0.init
crw-rw----  1 uucp  dialer    0x3f Feb 22 02:46 cuau0.lock
crw-rw----  1 uucp  dialer    0xcc Feb 22 03:45 cuaU0.lock
```

Plugged out:
```
root@HomeAssistant:~ # ls -l /dev | grep cua
crw-rw----  1 uucp  dialer    0x3d Feb 22 02:46 cuau0
crw-rw----  1 uucp  dialer    0x3e Feb 22 02:46 cuau0.init
crw-rw----  1 uucp  dialer    0x3f Feb 22 02:46 cuau0.lock
```

The think the dongle is "cuaU0"

Something that I observed. The green led of the dongle was previously always on. Now it turns off after 1-2min.

**Comment by paxswill** [comment]
Does the dongle show up within the jail as well (the `cuau0` entries shouldn't be visible within the jail, just the `cuaU` entries)? And if you turn up the [log level for zigbee2mqtt](https://www.zigbee2mqtt.io/information/debug.html#zigbee2mqtt-debug-logging) and restart it, does anything pop out?

**Comment by GitIgnoreMaybe** [comment]
I've set 

`advanced:
  log_level: debug`
  
  in the config, but where do I find the logs?

**Comment by paxswill** [comment]
They should be in the same place as before, `/var/log/zigbee2mqtt_daemon.log` (and `less -R` to make it pretty)

**Comment by paxswill** [comment]
They should also be visible in the web interface (port 8080).

**Comment by GitIgnoreMaybe** [comment]
That is the initial issue, I can't access it because of "refused to connect"

<img width="1180" alt="Bildschirmfoto 2021-02-23 um 00 00 52" src="https://user-images.githubusercontent.com/37028024/108781594-6ff14c80-756a-11eb-8caa-37111c95bbea.png">

**Comment by GitIgnoreMaybe** [comment]
This is the output of the logs

```
Last login: Mon Feb 22 23:59:08 on pts/27
FreeBSD 12.2-RELEASE-p3 7851f4a452d(HEAD) TRUENAS
^[[32mZigbee2MQTT:info ^[[39m 2021-01-30 18:38:02: Logging to console only'
^[[32mZigbee2MQTT:info ^[[39m 2021-01-30 18:38:02: Starting zigbee-herdsman (0.1
3.46)
^[[31mZigbee2MQTT:error^[[39m 2021-01-30 18:38:10: Error: Failed to connect to t
dist/adapter/z-stack/adapter/zStackAdapter.js:94:31)
^[[31mZigbee2MQTT:error^[[39m 2021-01-30 18:41:59: Error while starting zigbee-h
er/z-stack/znp/znp.js:136:32)
    at Generator.next (<anonymous>)
tack/znp/znp.js:23:12)
    at SerialPort._error (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/node_mod
^[[31mZigbee2MQTT:error^[[39m 2021-01-30 18:48:11: Error while starting zigbee-h
rialport 'Error: Error: Device not configured, cannot open /dev/cuaU1'
tack/znp/znp.js:23:12)
1.17.0 (commit #435a64a)
^[[32mZigbee2MQTT:info ^[[39m 2021-01-30 18:57:27: MQTT publish: topic 'homeassi
^[[32mZigbee2MQTT:info ^[[39m 2021-01-30 18:38:02: Logging to console only'
^[[32mZigbee2MQTT:info ^[[39m 2021-01-30 18:38:02: Starting zigbee-herdsman (0.1
3.46)
^[[31mZigbee2MQTT:error^[[39m 2021-01-30 18:38:10: Error: Failed to connect to t
dist/adapter/z-stack/adapter/zStackAdapter.js:94:31)
^[[31mZigbee2MQTT:error^[[39m 2021-01-30 18:41:59: Error while starting zigbee-h
er/z-stack/znp/znp.js:136:32)
    at Generator.next (<anonymous>)
tack/znp/znp.js:23:12)
    at SerialPort._error (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/node_mod
^[[31mZigbee2MQTT:error^[[39m 2021-01-30 18:48:11: Error while starting zigbee-h
rialport 'Error: Error: Device not configured, cannot open /dev/cuaU1'
tack/znp/znp.js:23:12)
1.17.0 (commit #435a64a)
1.17.0 (commit #435a64a)
^[[32mZigbee2MQTT:info ^[[39m 2021-01-30 18:38:02: Logging to console only'
^[[32mZigbee2MQTT:info ^[[39m 2021-01-30 18:38:02: Starting zigbee-herdsman (0.1
3.46)
^[[31mZigbee2MQTT:error^[[39m 2021-01-30 18:38:10: Error: Failed to connect to t
dist/adapter/z-stack/adapter/zStackAdapter.js:94:31)
^[[31mZigbee2MQTT:error^[[39m 2021-01-30 18:41:59: Error while starting zigbee-h
er/z-stack/znp/znp.js:136:32)
    at Generator.next (<anonymous>)
tack/znp/znp.js:23:12)
    at SerialPort._error (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/node_mod
^[[31mZigbee2MQTT:error^[[39m 2021-01-30 18:48:11: Error while starting zigbee-h
rialport 'Error: Error: Device not configured, cannot open /dev/cuaU1'
tack/znp/znp.js:23:12)
^[[32mZigbee2MQTT:info ^[[39m 2021-01-30 18:38:02: Logging to console only'
^[[32mZigbee2MQTT:info ^[[39m 2021-01-30 18:38:02: Starting Zigbee2MQTT version
1.17.0 (commit #435a64a)
^[[32mZigbee2MQTT:info ^[[39m 2021-01-30 18:38:02: Starting zigbee-herdsman (0.1
3.46)
^[[31mZigbee2MQTT:error^[[39m 2021-01-30 18:38:10: Error while starting zigbee-h
erdsman
^[[31mZigbee2MQTT:error^[[39m 2021-01-30 18:38:10: Failed to start zigbee
^[[31mZigbee2MQTT:error^[[39m 2021-01-30 18:38:10: Exiting...
^[[31mZigbee2MQTT:error^[[39m 2021-01-30 18:38:10: Error: Failed to connect to t
he adapter (Error: Cannot request when znp has not been initialized yet)
    at ZStackAdapter.<anonymous> (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/
dist/adapter/z-stack/adapter/zStackAdapter.js:94:31)
    at Generator.throw (<anonymous>)
    at rejected (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/dist/adapter/z-st
ack/adapter/zStackAdapter.js:25:65)
^[[32mZigbee2MQTT:info ^[[39m 2021-01-30 18:41:59: Logging to console only'
^[[32mZigbee2MQTT:info ^[[39m 2021-01-30 18:41:59: Starting Zigbee2MQTT version
1.17.0 (commit #435a64a)
^[[32mZigbee2MQTT:info ^[[39m 2021-01-30 18:41:59: Starting zigbee-herdsman (0.1
3.46)
...
```

before that, I can see some payloads
```
[[32mZigbee2MQTT:info ^[[39m 2021-01-30 18:49:39: Connected to MQTT server
^[[32mZigbee2MQTT:info ^[[39m 2021-01-30 18:49:39: MQTT publish: topic 'zigbee2m
qtt/bridge/state', payload 'online'
^[[32mZigbee2MQTT:info ^[[39m 2021-01-30 18:49:39: MQTT publish: topic 'zigbee2m
qtt/bridge/info', payload '{"commit":"435a64a","config":{"advanced":{"adapter_co
ncurrent":null,"adapter_delay":null,"availability_blacklist":[],"availability_bl
ocklist":[],"availability_passlist":[],"availability_timeout":0,"availability_wh
itelist":[],"cache_state":true,"cache_state_persistent":true,"cache_state_send_o
n_startup":true,"channel":11,"elapsed":false,"ext_pan_id":[221,221,221,221,221,2
21,221,221],"homeassistant_discovery_topic":"homeassistant","homeassistant_legac
y_triggers":true,"homeassistant_status_topic":"hass/status","last_seen":"disable
","legacy_api":true,"log_directory":"/usr/local/etc/zigbee2mqtt/log/%TIMESTAMP%"
,"log_file":"log.txt","log_level":"info","log_output":["console"],"log_rotation"
:true,"log_syslog":{},"pan_id":6754,"report":false,"soft_reset_timeout":0,"times
tamp_format":"YYYY-MM-DD HH:mm:ss"},"ban":[],"blocklist":[],"device_options":{},
"devices":{},"experimental":{"new_api":true,"output":"json"},"external_converter ...
```

**Comment by paxswill** [comment]
The logs say that it’s trying to connect to `cuaU1`, but your previous post says that your adapter is on `cuaU0`. Try changing that in the config file:
```
serial:
  port: /dev/cuaU0
```
I’m curious, do you have that currently set to `null` for auto detection?

**Comment by GitIgnoreMaybe** [comment]
These have been the previous settings:
```
# Home Assistant integration (MQTT discovery)
homeassistant: false

# allow new devices to join
permit_join: true

# MQTT settings
mqtt:
  # MQTT base topic for zigbee2mqtt MQTT messages
  base_topic: zigbee2mqtt
  # MQTT server URL
  server: 'mqtt://localhost'
  # MQTT server authentication, uncomment if required:
  # user: my_user
  # password: my_password

advanced:
  log_level: debug

# Serial settings
serial:
  # Location of CC2531 USB sniffer
  port: /dev/ttyACM0
```

I assume I need to restart the service. How can I do it? Just restarting via the Truenas restart button?

**Comment by GitIgnoreMaybe** [comment]
Seems like this is related:
https://github.com/Koenkk/zigbee2mqtt/issues/2928

**Comment by paxswill** [comment]
That shouldn’t be the config, [here’s the config](https://github.com/paxswill/iocage-zigbee2mqtt/blob/master/overlay/usr/local/etc/zigbee2mqtt/configuration.yaml) installed by the plug-in (it should be at `/usr/local/etc/zigbee2mqtt/configuration.yaml` within the jail). For restarting, you can do `service zigbee2mqtt restart` also within the jail (but doing the restart button within the TrueNAS UI should work too).

**Comment by GitIgnoreMaybe** [comment]
I see my prev. post was `/opt/zigbee2mqtt/data/configuration.yaml`. 

So `/usr/local/etc/zigbee2mqtt/configuration.yaml` has exactly the same settings as your link.

**Comment by paxswill** [comment]
The logs say that the config file is using `/dev/cuaU*1*`, but the default config is for cuaU0. I also just noticed those logs are also a few weeks old. If you haven’t already, try restarting the jail (the restart button in the TrueNAS UI) and then what do the most recent logs say (in `less`, type ‘G’ (shift-g) to go to the end of the file)?

**Comment by GitIgnoreMaybe** [comment]
Sorry, you are right I shared old logs.

```
Zigbee2MQTT:info  2021-02-22 00:42:21: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"Publish 'set' 'brightness' to 'Satellite' failed: 'Error: Command 0xbc33acfffe94e682/1 genLevelCtrl.moveToLevelWithOnOff({\"level\":74,\"transtime\":0}, {\"timeout\":10000,\"disableResponse\":false,\"disableRecovery\":false,\"disableDefaultResponse\":false,\"direction\":0,\"srcEndpoint\":null,\"reservedBits\":0,\"manufacturerCode\":null,\"transactionSequenceNumber\":null}) failed (SRSP - AF - dataRequest after 6000ms)'","meta":{"friendly_name":"Satellite"},"type":"zigbee_publish_error"}'
Zigbee2MQTT:error 2021-02-22 00:42:27: Publish 'set' 'brightness' to 'meeting-left' failed: 'Error: Command 0x5c0272fffe38d3e6/1 genLevelCtrl.moveToLevelWithOnOff({"level":248,"transtime":0}, {"timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":false,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null}) failed (SRSP - AF - dataRequest after 6000ms)'
Zigbee2MQTT:info  2021-02-22 00:42:27: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"Publish 'set' 'brightness' to 'meeting-left' failed: 'Error: Command 0x5c0272fffe38d3e6/1 genLevelCtrl.moveToLevelWithOnOff({\"level\":248,\"transtime\":0}, {\"timeout\":10000,\"disableResponse\":false,\"disableRecovery\":false,\"disableDefaultResponse\":false,\"direction\":0,\"srcEndpoint\":null,\"reservedBits\":0,\"manufacturerCode\":null,\"transactionSequenceNumber\":null}) failed (SRSP - AF - dataRequest after 6000ms)'","meta":{"friendly_name":"meeting-left"},"type":"zigbee_publish_error"}'
Zigbee2MQTT:info  2021-02-22 00:42:42: Logging to console only'
Zigbee2MQTT:info  2021-02-22 00:42:42: Starting Zigbee2MQTT version 1.17.0 (commit #435a64a)
Zigbee2MQTT:info  2021-02-22 00:42:42: Starting zigbee-herdsman (0.13.46)
Zigbee2MQTT:error 2021-02-22 00:43:02: Error while starting zigbee-herdsman
Zigbee2MQTT:error 2021-02-22 00:43:02: Failed to start zigbee
Zigbee2MQTT:error 2021-02-22 00:43:02: Exiting...
Zigbee2MQTT:error 2021-02-22 00:43:02: Error: Failed to connect to the adapter (Error: SRSP - SYS - ping after 6000ms)
    at ZStackAdapter.<anonymous> (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/dist/adapter/z-stack/adapter/zStackAdapter.js:94:31)
    at Generator.throw (<anonymous>)
    at rejected (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/dist/adapter/z-stack/adapter/zStackAdapter.js:25:65)
Zigbee2MQTT:info  2021-02-22 00:44:37: Logging to console only'
Zigbee2MQTT:info  2021-02-22 00:44:37: Starting Zigbee2MQTT version 1.17.0 (commit #435a64a)
Zigbee2MQTT:info  2021-02-22 00:44:37: Starting zigbee-herdsman (0.13.46)
Zigbee2MQTT:error 2021-02-22 00:44:57: Error while starting zigbee-herdsman
Zigbee2MQTT:error 2021-02-22 00:44:57: Failed to start zigbee
Zigbee2MQTT:info  2021-02-22 00:42:21: MQTT publish: topic 'zigbee2mqtt/bridge/l
og', payload '{"message":"Publish 'set' 'brightness' to 'Satellite' failed: 'Err
or: Command 0xbc33acfffe94e682/1 genLevelCtrl.moveToLevelWithOnOff({\"level\":74
,\"transtime\":0}, {\"timeout\":10000,\"disableResponse\":false,\"disableRecover
y\":false,\"disableDefaultResponse\":false,\"direction\":0,\"srcEndpoint\":null,
\"reservedBits\":0,\"manufacturerCode\":null,\"transactionSequenceNumber\":null}
) failed (SRSP - AF - dataRequest after 6000ms)'","meta":{"friendly_name":"Satel
lite"},"type":"zigbee_publish_error"}'
Zigbee2MQTT:error 2021-02-22 00:42:27: Publish 'set' 'brightness' to 'meeting-le
ft' failed: 'Error: Command 0x5c0272fffe38d3e6/1 genLevelCtrl.moveToLevelWithOnO
ff({"level":248,"transtime":0}, {"timeout":10000,"disableResponse":false,"disabl
eRecovery":false,"disableDefaultResponse":false,"direction":0,"srcEndpoint":null
,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null}) fai
led (SRSP - AF - dataRequest after 6000ms)'
Zigbee2MQTT:info  2021-02-22 00:42:27: MQTT publish: topic 'zigbee2mqtt/bridge/l
og', payload '{"message":"Publish 'set' 'brightness' to 'meeting-left' failed: '
Error: Command 0x5c0272fffe38d3e6/1 genLevelCtrl.moveToLevelWithOnOff({\"level\"
:248,\"transtime\":0}, {\"timeout\":10000,\"disableResponse\":false,\"disableRec
overy\":false,\"disableDefaultResponse\":false,\"direction\":0,\"srcEndpoint\":n
ull,\"reservedBits\":0,\"manufacturerCode\":null,\"transactionSequenceNumber\":n
ull}) failed (SRSP - AF - dataRequest after 6000ms)'","meta":{"friendly_name":"m
eeting-left"},"type":"zigbee_publish_error"}'
Zigbee2MQTT:info  2021-02-22 00:42:42: Logging to console only'
Zigbee2MQTT:info  2021-02-22 00:42:42: Starting Zigbee2MQTT version 1.17.0 (commit #435a64a)
Zigbee2MQTT:info  2021-02-22 00:42:42: Starting zigbee-herdsman (0.13.46)
Zigbee2MQTT:error 2021-02-22 00:43:02: Error while starting zigbee-herdsman
Zigbee2MQTT:error 2021-02-22 00:43:02: Failed to start zigbee
Zigbee2MQTT:error 2021-02-22 00:43:02: Exiting...
Zigbee2MQTT:error 2021-02-22 00:43:02: Error: Failed to connect to the adapter (Error: SRSP - SYS - ping after 6000ms)
    at ZStackAdapter.<anonymous> (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/dist/adapter/z-stack/adapter/zStackAdapter.js:94:31)
    at Generator.throw (<anonymous>)
    at rejected (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/dist/adapter/z-stack/adapter/zStackAdapter.js:25:65)
Zigbee2MQTT:info  2021-02-23 00:57:01: Logging to console only'
Zigbee2MQTT:info  2021-02-23 00:57:01: Starting Zigbee2MQTT version 1.17.0 (commit #435a64a)
Zigbee2MQTT:info  2021-02-23 00:57:01: Starting zigbee-herdsman (0.13.46)
Zigbee2MQTT:error 2021-02-23 00:57:01: Failed to start zigbee
Zigbee2MQTT:error 2021-02-23 00:57:01: Exiting...
Zigbee2MQTT:error 2021-02-23 00:57:01: Error: EACCES: permission denied, open '/usr/local/etc/zigbee2mqtt/configuration.yaml'
    at Object.openSync (node:fs:495:3)
    at Object.writeFileSync (node:fs:1519:35)
    at Object.writeIfChanged (/opt/zigbee2mqtt/lib/util/yaml.js:24:12)
    at write (/opt/zigbee2mqtt/lib/util/settings.js:349:10)
    at Object.set (/opt/zigbee2mqtt/lib/util/settings.js:543:5)
    at Zigbee.start (/opt/zigbee2mqtt/lib/zigbee.js:52:22)
    at processTicksAndRejections (node:internal/process/task_queues:93:5)
    at async Controller.start (/opt/zigbee2mqtt/lib/controller.js:111:13)
(END)
```

**Comment by GitIgnoreMaybe** [comment]
I double-checked the config as well. From where can the `/dev/cuaU*1*` come from?
<img width="609" alt="Bildschirmfoto 2021-02-23 um 01 36 28" src="https://user-images.githubusercontent.com/37028024/108788140-98cc0e80-7577-11eb-8582-828fc888dcf9.png">

**Comment by paxswill** [comment]
Hmmm, the cuaU1 might’ve been something you tried earlier, there isn’t a reference to it in the latest logs. The two things that are jumping out to me are

1. The line:
    ```
    Error: Failed to connect to the adapter (Error: SRSP - SYS - ping after 6000ms)
    ```
   turns up a [couple results](https://github.com/Koenkk/zigbee2mqtt/issues/2928) that mention reflashing the dongle helped (there’s also some possibly relevant steps in the later parts of the [upstream FAQ](https://www.zigbee2mqtt.io/information/FAQ.html#help-zigbee2mqtt-fails-to-start) might help.

2. The permission error is suspicious. The permissions for everything in `/usr/local/etc/zigbee2mqtt` should be at least user r/w, with the owning user `zigbee2mqtt`. GENERATE still being in there is also weird, as as soon as zigbee2mqtt starts it should replace that with the newly generated key.

**Comment by GitIgnoreMaybe** [comment]
Thanks so far! I'm going to order a new dongle and check if this changes anything.

**Comment by GitIgnoreMaybe** [comment]
I got the new dongle. The new one has the same error. Any idea what could be the next option?

**Comment by GitIgnoreMaybe** [comment]
Alright, I got it up running again. But all devices are gone. This is what I did:

1. Went into the zigbee2mqtt jail
2. Changed the permission with `chmod 777 /
usr/local/etc/zigbee2mqtt/configuration.yaml`
2. Changed the mqtt server in ` /usr/local/etc/zigbee2mqtt/configuration.yaml` from local to the IP-ADDRESS
3. Pressed the reset button described here: https://github.com/Koenkk/zigbee2mqtt/issues/1597#issuecomment-500043780
4. Restarted `service zigbee2mqtt restart`

The great thing is that I'm able to access the service again. But the table is blank and all devices are gone. Is there a way to restore them?
